db/migrate/20190304192641_add_provider_guids.rb

```diff
@@ -0,0 +1,11 @@
+class AddProviderGuids < ActiveRecord::Migration[5.0]
+  class Provider < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  def up
+    Provider.where(:guid => nil).each { |provider| provider.update_attributes!(:guid => SecureRandom.uuid) }
```

Comment:
This looks like it's solving a different problem than uniqueness ... Can this be a separate PR or are they somehow related?

Reply:
Also you added the mixin for regions, but are not using `.in_my_region` here ...